WPMedia/muckrock
# NE-394 Generate presigned url for async files

**Summary**: Since we are using a private s3 bucket to store user-uploaded files, including zipped versions of request text and csv summaries of requests, we need to provide django's templates with a signed url to access the files on s3. This updates the `AsyncFileDownload` class which is subclassed by the zip and csv models to provide a [presigned s3 url](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.generate_presigned_url) and pass it down to the templates. We bumped the expiration to 30 days, since the buckets now have a corresponding lifecycle policy that expires objects after 30 days. 

**Ticket**: https://arcpublishing.atlassian.net/browse/NE-394

**Testing**: 
Start both muckrock and squarelet locally, making sure that your `DJANGO_SETTINGS_MODULE=muckrock.settings.wapo.local` and your `MAILGUN_SERVER_NAME=dev.muckrock.com` is set in `.envs/.local/.django`. 

1. Navigate to dev.muckrock.com and generate a foia request
2. Click `Download as Zip` and verify that the file downloads
3. go to http://dev.muckrock.com/foi/mylist/ and click `Export Search Results as CSV`
4. go to `dev.mailhog.com` and click the refresh arrow button, you should see 2 emails, each with a pre-signed link to a s3 object. Verify that both links work. 

**Screenshot**: 
Your email should look like:

![Screen Shot 2021-05-13 at 1 54 57 PM](https://user-images.githubusercontent.com/6980388/118166832-f859ed80-b3f3-11eb-9466-afd47869e808.png)
